woocommerce/woocommerce-gateway-stripe
# Add order lock for legacy redirect payments

Fixes #1302 

In the legacy redirect payment flow, no order locking mechanism has been implemented. This might cause double order note/email issue. This PR prevents the possible double order note/email issue by imposing order locking during redirect payments to avoid processing the same order by webhook as well.

## Changes proposed in this Pull Request:
- Added order lock to the redirect payment flow for legacy checkout.

## Testing instructions
- Enable legacy checkout experience.
- Enable a few other payment methods apart from the card (i,e iDeal, EPS, Bancontact) that need redirection.
- As a shopper, make purchases with the other payment methods. Confirm that the payments work fine.
